crates/python/src/execution_data.rs

```diff
@@ -46,35 +51,53 @@ py_wrap_data_struct! {
         duration: Option<Duration> => Option<Py<PyDelta>>
     }
 }
+impl_repr!(PyExecutionData);
 
 #[pymethods]
 impl PyExecutionData {
     #[new]
-    fn __new__(py: Python<'_>, result_data: PyResultData, duration: Option<u64>) -> PyResult<Self> {
+    fn __new__(
+        py: Python<'_>,
+        result_data: PyResultData,
+        duration: Option<Py<PyDelta>>,
+    ) -> PyResult<Self> {
         Ok(Self(ExecutionData {
             result_data: ResultData::py_try_from(py, &result_data)?,
-            duration: duration.map(Duration::from_micros),
+            duration: match duration {
+                None => None,
+                Some(delta) => Some(
+                    delta
+                        .as_ref(py)
+                        .call_method0("total_seconds")
+                        .map(|result| result.extract::<f64>())?
+                        .map(Duration::from_secs_f64)?,
+                ),
+            },
```

Comment:
Suggesting mostly as something that would maybe trip up clippy.

```suggestion
            duration: duration.map(|delta| {
                delta
                    .as_ref(py)
                    .call_method0("total_seconds")
                    .map(|result| result.extract::<f64>())?
                    .map(Duration::from_secs_f64)
            }).transpose()?,
```

Reply:
Why is the `transpose()` preferable to having the `?` after the last line of the closure?

I agree with keeping the `duration.map(|delta|...)` rather than the explicit `match`.

Reply:
I rewrote it as suggested, but left off the `transpose`